common/ipfs/api.go

```diff
@@ -0,0 +1,99 @@
+package ipfs
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/abevier/tsk/ratelimiter"
+
+	iface "github.com/ipfs/boxo/coreiface"
+	"github.com/ipfs/kubo/client/rpc"
+	ma "github.com/multiformats/go-multiaddr"
+
+	"github.com/ceramicnetwork/go-cas/models"
+)
+
+const defaultIpfsRateLimit = 16
+const defaultIpfsBurstLimit = 16
+const defaultIpfsLimiterMaxQueueDepth = 100
+const defaultIpfsPublishPubsubTimeoutS = 30
+
+type PubSubPublishTask struct {
```

Comment:
Want to move this to a new file where we can add more tasks in the future? `task.go`?